docs_src/design/ucr/Global Common Configuration.md

```diff
@@ -0,0 +1,169 @@
+
+!!! note
+    In Levski release App Services don't have the **MessageQueue** section and Core Command's is **MessageQueue.Internal**. These inconstancies will be rectified in EdgeX 3.0 so all EdgeX services have the same **MessageQueue** section specified in the same manner. Also in EdgeX 3.0, the **PublishTopicPrefix** and **SubscribeTopic** settings will be replaced by entries in **MessageQueue.Topics**.
+
+There are other similar common sections not shown shown above. As can be seen from the two examples above there is much duplication of configuration settings across all the EdgeX services. This gives rise to the need to have all these common duplicate configuration settings in a single global source. 
+
+### Existing solutions
+There are no existing solutions for global configuration for EdgeX since the current configuration implementation is specific to EdgeX.
+
+### Requirements
+
+#### General
+
+- Services shall be able reference a global common configuration in a manner that is flexible for use with and without the Service Provider
+- Common environment overrides shall be applied to global common configuration settings only once before the services receive the setting values
+
+    - This avoids the need for duplicate environment overrides on each service in the docker compose files
+
+- Services must be able to override any of the global common configuration settings with local service specific TOML values
+
+    !!! example - "Example Core Data specific **Writable.Telemetry** and **Service** configuration settings in local TOML file"
+        ```toml
+          [Writable.Telemetry]
+            [Writable.Telemetry.Metrics] # All service's metric names must be present in this list.
+            EventsPersisted = false
+            ReadingsPersisted = false
+
+        ...
+        
+        [Service]
+        Port = 59880
+        StartupMsg = "This is the Core Data Microservice"
+        ```
+
+- Environment variable overrides shall be applied in the same manner as they are today. 
+    - After loaded from file (prior to pushing into the the Configuration Provider) 
+    - After loaded from the Configuration Provider
+
+#### With Configuration Provider
+
+- Global common configuration shall be pre-loaded into the Configuration Provider where it is pulled from by each service.
+    - How and where the configuration is loaded into the Configuration Provider is left to design
+- Pre-loading of global common configuration must have a mechanism to indicate that the configuration pre-load is complete 
+- Each service must wait until the global common configuration pre-load is complete.
+- Each service's full configuration, global common and local TOML overrides, shall be pushed into the Configuration Provider
+    - Once initially loaded the service's full configuration is solely referenced from the Configuration Provider as it is today

```

Comment:
I am suggesting that it isn't done that way.

Reply:
The configuration would not be loaded into the services structure prior to pushing to the Config Provider. Only when pulled from the Config Provider. This then complicates making a change in config provider to override a global that doesn't exist in the service's section. It can be done, but users would have to know how to properly add the override in the Config Provider UI or REST API. 

Something for the ADR discussion.